Season-2/Level-3/hack.txt

```diff
@@ -0,0 +1,6 @@
+Here are two steps to simulate an attack:
+
```

Comment:
Whenever we can have an automated test instead of a manual one, let's do it, and here was an example of having automated ones

Reply:
Nope, the original `hack.py` doesn't demonstrate anything in automated fashion, as it is hitting the programmatic API and printing the result onto the screen. Here is the dump from the session:
```
@evarga ➜ /workspaces/secure-code-game (main) $ python3 Season-2/Level-3/hack.py
<h2>Planet Details:</h2><p>No information found for <<img src='x' onerror='alert(1)'>>.</p>
```
Honestly, I don't see anything failing here nor an XSS attack in action. For the latter to happen, rendering must be performed over the response to actually execute any embedded active content.

Again, the fact that those `<<` and `>>` symbols are shown is just due to the blunder in the `code.py`, as I've explained in #62.

P.S. It is also a very bad practice to return form an API endpoint a stylized HTML content. This is another reason why I've decided to completely omit it from the revised version, as it would even more confuse students where to look for issues.

Reply:
Makes sense now, thanks for pointing out. See my comment above and if you can re-write your proposal following what I suggested then it's a win-win. In addition, feel free to either try to automate the hack file or keep it manual, making sure of course that reflects the changes on the code file.

Reply:
Sure, I will alter the code accordingly for attaining a desirable win-win state.